python/ray/serve/_private/proxy_state.py

```diff
@@ -121,66 +266,53 @@ def _health_check(self):
         """Perform periodic health checks."""
         assert self._status in {ProxyStatus.HEALTHY, ProxyStatus.DRAINING}
 
-        if self._health_check_obj_ref:
-            finished, _ = ray.wait([self._health_check_obj_ref], timeout=0)
-            if finished:
-                self._health_check_obj_ref = None
-                try:
-                    ray.get(finished[0])
-                    # Call to reset _consecutive_health_check_failures
-                    # the status should be unchanged.
+        if self._proxy_actor_wrapper.health_check_obj_ref:
+            try:
+                healthy = self._proxy_actor_wrapper.is_healthy()
+                if healthy:
                     self.try_update_status(self._status)
-                except ray.exceptions.RayActorError:
-                    # The proxy actor dies.
-                    self.set_status(ProxyStatus.UNHEALTHY)
-                except Exception as e:
+                elif (
+                    time.time() - self._last_health_check_time
+                    > PROXY_HEALTH_CHECK_TIMEOUT_S
+                ):
+                    # Health check hasn't returned and the timeout is up, consider it
+                    # failed.
+                    self._proxy_actor_wrapper.clear_health_check_obj_ref()
                     logger.warning(
-                        f"Health check for proxy {self._actor_name} failed: {e}"
+                        "Didn't receive health check response for proxy "
+                        f"{self._node_id} after {PROXY_HEALTH_CHECK_TIMEOUT_S}s"
                     )
                     self.try_update_status(ProxyStatus.UNHEALTHY)
-            elif (
-                time.time() - self._last_health_check_time
-                > PROXY_HEALTH_CHECK_TIMEOUT_S
-            ):
-                # Health check hasn't returned and the timeout is up, consider it
-                # failed.
-                self._health_check_obj_ref = None
-                logger.warning(
-                    "Didn't receive health check response for proxy "
-                    f"{self._node_id} after {PROXY_HEALTH_CHECK_TIMEOUT_S}s"
-                )
+            except RayActorError:
+                # The proxy actor dies.
+                self.set_status(ProxyStatus.UNHEALTHY)
+            except Exception as e:
+                logger.warning(f"Health check for proxy {self._actor_name} failed: {e}")
                 self.try_update_status(ProxyStatus.UNHEALTHY)
 
         # If there's no active in-progress health check and it has been more than 10
         # seconds since the last health check, perform another health check.
-        if self._health_check_obj_ref:
+        if self._proxy_actor_wrapper.health_check_obj_ref:
```

Comment:
The obj_ref internal implementation detail should not be exposed. Instead, add a method like "health check ongoing"